packer/builder/azure/arm/builder.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See the LICENSE file in the project root for license information.
+
+package arm
+
+import (
+	"log"
+
+	"github.com/Azure/packer-azure/packer/builder/azure/common/constants"
+	"github.com/Azure/packer-azure/packer/builder/azure/common/lin"
+
+	"github.com/Azure/azure-sdk-for-go/Godeps/_workspace/src/github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/mitchellh/multistep"
+	"github.com/mitchellh/packer/common"
+	"github.com/mitchellh/packer/helper/communicator"
+	"github.com/mitchellh/packer/packer"
+)
+
+type Builder struct {
+	config   *Config
+	stateBag multistep.StateBag
+	runner   multistep.Runner
+}
+
+const (
+	DefaultPublicIPAddressName = "sshPublicIP"
+)
+
+func (b *Builder) Prepare(raws ...interface{}) ([]string, error) {
+	c, warnings, errs := newConfig(raws...)
+	if errs != nil {
+		return warnings, errs
+	}
+
+	b.config = c
+
+	b.stateBag = new(multistep.BasicStateBag)
+	err := b.configureStateBag(b.stateBag)
+	if err != nil {
+		return nil, err
+	}
+
+	return warnings, errs
+}
+
+func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packer.Artifact, error) {
+	ui.Say("Preparing builder ...")
+	ui.Message("Creating Azure Resource Manager (ARM) client ...")
```

Comment:
neater to move this closer to the actual creation call (of either the principal or the actual client)